receivers/email_sender.go

```diff
@@ -0,0 +1,242 @@
+package receivers
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	_ "embed"
+	"fmt"
+	"html/template"
+	"net"
+	"net/mail"
+	"strconv"
+	"strings"
+
+	"github.com/Masterminds/sprig/v3"
+	gomail "gopkg.in/mail.v2"
+)
+
+//go:embed templates/ng_alert_notification.html
+var defaultEmailTemplate string
+
+type EmailSenderConfig struct {
+	AuthPassword  string
+	AuthUser      string
+	CertFile      string
+	EhloIdentity  string
+	ExternalURL   string
+	FromName      string
+	FromAddress   string
+	Host          string
+	KeyFile       string
+	SkipVerify    bool
+	StaticHeaders map[string]string
+	Version       string
+}
+
+type defaultEmailSender struct {
+	cfg  EmailSenderConfig
+	tmpl *template.Template
+}
+
+// NewEmailSenderFactory takes a configuration and returns a new EmailSender factory function.
+func NewEmailSenderFactory(cfg EmailSenderConfig) func(Metadata) (EmailSender, error) {
+	return func(n Metadata) (EmailSender, error) {
+		tmpl, err := template.New("ng_alert_notification").
+			Funcs(template.FuncMap{
+				"Subject":                 subjectTemplateFunc,
+				"__dangerouslyInjectHTML": __dangerouslyInjectHTML,
+			}).
+			Funcs(sprig.FuncMap()).
+			Parse(defaultEmailTemplate)
+		if err != nil {
+			return nil, err
+		}
+
+		return &defaultEmailSender{
+			cfg:  cfg,
+			tmpl: tmpl,
+		}, nil
+	}
+}
+
+// Message representats an email message.
+type Message struct {
+	To            []string
+	From          string
+	Subject       string
+	Body          string
+	EmbeddedFiles []string
+	ReplyTo       []string
+	SingleEmail   bool
+}
+
+// SendEmail implements the EmailSender interface.
+func (s *defaultEmailSender) SendEmail(_ context.Context, cmd *SendEmailSettings) error {
+	message, err := s.buildEmailMessage(cmd)
+	if err != nil {
+		return err
+	}
+
+	_, err = s.Send(message)
+	return err
+}
+
+func (s *defaultEmailSender) buildEmailMessage(cmd *SendEmailSettings) (*Message, error) {
+	data := cmd.Data
+	if data == nil {
+		data = make(map[string]any, 10)
+	}
+
+	s.setDefaultTemplateData(data)
+
+	var buffer bytes.Buffer
+	if err := s.tmpl.ExecuteTemplate(&buffer, cmd.Template, data); err != nil {
+		return nil, err
+	}
+
+	subject := cmd.Subject
+	if cmd.Subject == "" {
+		subjectData := data["Subject"].(map[string]any)
+		subjectText, hasSubject := subjectData["executed_template"].(string)
+		if !hasSubject {
+			return nil, fmt.Errorf("missing subject in template %s", cmd.Template)
+		}
+		subject = subjectText
+	}
+
+	addr := mail.Address{Name: s.cfg.FromName, Address: s.cfg.FromAddress}
+	return &Message{
+		To:            cmd.To,
+		From:          addr.String(),
+		Subject:       subject,
+		Body:          buffer.String(),
+		EmbeddedFiles: cmd.EmbeddedFiles,
+		ReplyTo:       cmd.ReplyTo,
+		SingleEmail:   cmd.SingleEmail,
+	}, nil
+}
+
+func (s *defaultEmailSender) setDefaultTemplateData(data map[string]any) {
+	data["AppUrl"] = s.cfg.ExternalURL
+	data["BuildVersion"] = s.cfg.Version
```

Comment:
We don't have `BuildStamp`, `EmailCodeValidHours` and `Name` here, but they're not used in our template so that makes sense.